frontend/src/app/offers/offer-detail/offer-detail.component.ts

```diff
@@ -0,0 +1,29 @@
+import { environment } from '../../../environments/environment';
+import { observable } from 'rxjs/symbol/observable';
+import { Offer } from '../offers.model';
+import { OffersService } from '../offers.service';
+import { ActivatedRoute } from '@angular/router';
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/switchMap';
+
+@Component({
+  selector: 'volontulo-app-offer-detail',
+  templateUrl: './offer-detail.component.html',
+  styleUrls: ['./offer-detail.component.css'],
+})
+export class OfferDetailComponent implements OnInit {
+  public offer$: Observable<Offer>;
+  public djangoRoot = environment.djangoRoot;
+  public getDjangoViewUrl = this.offersService.getDjangoViewUrl;
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private offersService: OffersService,
+  ) { }
+
+  ngOnInit() {
+    this.offer$ = this.activatedRoute.params
```

Comment:
I think it would be a much cleaner solution of OfferDetailComponent accepted `offer` as input. Then you could reuse it on a list of components. With the current setup, on a list of components you'd need to first get a list of all IDs you wish to present, pass it to detail components and each component would then query for details.

Angular allows you to fetch data before loading a component in router. Check this out:
docs: https://angular.io/api/router/Resolve
tutorial: https://angular.io/guide/router#resolve-guard

Reply:
Offer detail component is not likely to be the same as the one on the listing, as it's much more verbose. This is a single offer page.

Reply:
OK